components/widgets/menu/menu_wrapper.tsx

```diff
@@ -68,6 +68,10 @@ export default class MenuWrapper extends React.PureComponent<Props, State> {
     }
 
     private keyboardClose = (e: KeyboardEvent) => {
+        if (this.state.open) {
+            e.stopPropagation();
+        }
```

Comment:
I think this one should probably be moved into the if block below just to avoid swallowing other key presses that we might want handled while the menu is still open